pkg/app/app.go

```diff
@@ -27,18 +25,26 @@ func initConfiguration(config configuration.Configuration, apiClient api.ApiClie
 	config.AddDefaultValue(configuration.CACHE_PATH, configuration.StandardDefaultValueFunction(dir))
 
 	config.AddDefaultValue(configuration.API_URL, func(existingValue any) any {
-		if existingValue == nil {
-			return constants.SNYK_DEFAULT_API_URL
-		} else {
-			apiString := existingValue
+		urlString := constants.SNYK_DEFAULT_API_URL
+
+		if existingValue != nil {
 			if temp, ok := existingValue.(string); ok {
-				if apiUrl, err := url.Parse(temp); err == nil {
-					apiUrl.Path = strings.Replace(apiUrl.Path, "/v1", "", 1)
-					apiString = apiUrl.String()
-				}
+				urlString = temp
 			}
-			return apiString
 		}
+
+		apiString, _ := api.GetCanonicalApiUrl(urlString)
```

Comment:
why is the error ignored here?

Reply:
Unfortunately we currently don't have error reporting or logging capabilities here. I would like to add this in a new PR if this is okay with you.

Reply:
+1
Also I don't know if there's a way to recover from this error, so it's just a matter of failing/logging an error earlier